src/views/AddItem.jsx

```diff
@@ -1,3 +1,66 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { useState } from 'react';
+import { addItem } from '../api/firebase';
+
+const defaultItem = { itemName: '', daysUntilNextPurchase: 7 };
+export function AddItem({ listToken }) {
+	const [item, setItem] = useState(defaultItem);
+	const [status, setStatus] = useState('');
+
+	const updateItem = (e) => {
+		if (status) setStatus('');
+		if (
+			e.target.name !== 'daysUntilNextPurchase' &&
+			e.target.name !== 'itemName'
+		)
+			return;
+		let updateVal = e.target.value;
+		if (e.target.name === 'daysUntilNextPurchase') {
+			updateVal = Number(updateVal);
+		}
+		setItem({ ...item, [e.target.name]: updateVal });
+	};
+
+	const addItemToDatabase = (e) => {
+		e.preventDefault();
+
+		addItem(listToken, item)
+			.then(() => setStatus('Item added successfully!'))
+			.then(() => setItem(defaultItem))
+			.catch(() => setStatus('Item could not be added.'));
+	};
+
+	return (
+		<div>
+			<p>
+				Hello from the <code>/add-item</code> page!
+			</p>
+			<form onSubmit={addItemToDatabase}>
+				<label htmlFor="item">
```

Comment:
Clicking on "Add Item" label doesn't focus on the input, while clicking on "Select Frequency" does focus on the drop down. Maybe just updating value of htmlFor to match input id?

Reply:
Nice catch here thanks!